pkgs/by-name/mu/multipass/gui.nix

```
@@ -0,0 +1,75 @@
+{
+  meta,
+  multipass_src,
+  multipassd,
+  version,
+
+  flutter327,
+  gtkmm3,
+  kdePackages,
+  keybinder3,
+  lib,
+  libayatana-appindicator,
+  libnotify,
+  protobuf,
+  protoc-gen-dart,
+}:
+flutter327.buildFlutterApplication {
+  inherit version meta;
+  pname = "multipass-gui";
+  src = multipass_src;
+
+  sourceRoot = "${multipass_src.name}/src/client/gui";
+
+  pubspecLock = lib.importJSON ./pubspec.lock.json;
+
+  gitHashes = {
+    dartssh2 = "sha256-2pypKwurziwGLZYuGaxlS2lzN3UvJp3bRTvvYYxEqRI=";
+    hotkey_manager_linux = "sha256-aO0h94YZvgV/ggVupNw8GjyZsnXrq3qTHRDtuhNv3oI=";
+    system_info2 = "sha256-fly7E2vG+bQ/+QGzXk+DYba73RZccltdW2LpZGDKX60=";
+    tray_menu = "sha256-riiAiBEms+9ARog8i+MR1fto1Yqx+gwbBWyNbNq6VTM=";
+    window_size = "sha256-71PqQzf+qY23hTJvcm0Oye8tng3Asr42E2vfF1nBmVA=";
+    xterm = "sha256-h8vIonTPUVnNqZPk/A4ZV7EYCMyM0rrErL9ZOMe4ZBE=";
+  };
+
+  buildInputs = [
+    gtkmm3
+    kdePackages.qtbase
+    kdePackages.qtwayland
+    keybinder3
+    libayatana-appindicator
+    libnotify
+    protobuf
+    protoc-gen-dart
+  ];
+
+  nativeBuildInputs = [
+    kdePackages.wrapQtAppsHook
+  ];
+
+  preBuild = ''
+    mkdir -p lib/generated
+
+    # Generate the Dart gRPC code for the Multipass GUI.
+    ${protobuf}/bin/protoc \
+      --plugin=protoc-gen-dart=${lib.getExe protoc-gen-dart} \
+      --dart_out=grpc:lib/generated \
+      -I ${multipass_src}/src/rpc \
+      ${multipass_src}/src/rpc/multipass.proto \
+      google/protobuf/timestamp.proto
+  '';
+
+  postFixup = ''
+    mv $out/bin/multipass_gui $out/bin/multipass.gui
+
+    substituteInPlace $out/share/applications/multipass.gui.autostart.desktop \
+      --replace-fail "Exec=multipass.gui" "Exec=$out/bin/multipass.gui" \
+      --replace-fail "Icon=multipass.gui" "Exec=$out/app/multipass-gui/data/flutter_assets/assets/icon.png" \
+
+    cp $out/share/applications/multipass.gui.autostart.desktop $out/share/applications/multipass.gui.desktop
+  '';
+
+  extraWrapProgramArgs = ''
+    --prefix LD_LIBRARY_PATH : '${multipassd}/lib'
+  '';
```

```suggestion
  '';
  
  meta = {
    inherit (meta) homepage license maintainers platforms;
  };
```
While much more annoying, this keeps attributes like `meta.pos` intact for use with `nix edit`, etc

---

Fixed, thanks :)